src/apiProvider.ts

```diff
@@ -24,10 +24,16 @@ export class ApiProvider implements IApiProvider {
    */
    constructor(url: string, config?: AxiosRequestConfig) {
     this.url = url;
-    this.config = config || {
+    this.config = Object.assign({}, config, {
       timeout: 1000,
-    };
-}
+      // See: https://github.com/axios/axios/issues/983 regarding transformResponse
+      transformResponse: [
+        function(data: any) {
```

Comment:
Typescript compiling complained about a lack of type for data, so I put "any"

It seems in most cases it's a string, but looking at axios itself, it seems there are times that even axios needs to fall back when data is not a string.  

https://github.com/axios/axios/blob/master/lib/defaults.js#L72

At least for elrond cases, the config scoped for the purposes of elrond APIs seems to be able to assume a string.

What do you think?

Reply:
Hey there. It's ok for now to leave it with `any`. Regardless of what `axios` provides in `transformResponse` as `data`, I think we should make sure ourselves it's a string before passing it to `JSONbig.parse`. But this is out of the scope of this PR, I'll make sure we update the error handling here in the case `data` is not a string.

Reply:
@pctj101 I changed the base to development. Can you please resolve the two conflicts? They're small, you should just keep your changes :)

Reply:
@ccorcoveanu Hello again!  Appreciate the feedback.  Merged against development and matched up the indent.